masonry/src/render_root.rs

```diff
@@ -202,12 +202,14 @@ impl RenderRoot {
                 // See https://github.com/linebender/druid/issues/85 for discussion.
                 let last = self.last_anim.take();
                 let elapsed_ns = last.map(|t| now.duration_since(t).as_nanos()).unwrap_or(0) as u64;
-                let root_state = self.root_state();
-                if root_state.request_anim {
-                    root_state.request_anim = false;
-                    self.root_lifecycle(LifeCycle::AnimFrame(elapsed_ns));
-                    self.last_anim = Some(now);
-                }
+
+                run_update_anim_pass(self, elapsed_ns);
+
+                let mut root_state = self.widget_arena.get_state_mut(self.root.id()).item.clone();
+                self.post_event_processing(&mut root_state);
+
+                self.last_anim = Some(now);
```

Comment:
I don't think this will have the behaviour we want. Consider for example:
```patch
diff --git a/xilem/examples/variable_clock.rs b/xilem/examples/variable_clock.rs
index 8bead4d80..ea7dd599f 100644
--- a/xilem/examples/variable_clock.rs
+++ b/xilem/examples/variable_clock.rs
@@ -60,7 +60,7 @@ fn app_logic(data: &mut Clocks) -> impl WidgetView<Clocks> {
                     };
                 }
             },
-            |data: &mut Clocks, ()| data.now_utc = OffsetDateTime::now_utc(),
+            |data: &mut Clocks, ()| data.weight = (data.weight + 100.) % 1000.,
         ),
     )
 }
```
This will not animate smoothly, but will instead to jump to the full value, because when the animation frame occurs, the time will be the time since the last frame, rather than `0` as it is before this PR.

Reply:
Would you mind taking responsibility for this PR, then? It seems like this is more in your wheelhouse than mine, if you're currently working on frame pacing.

Reply:
Can do. I think this only needs a very small change to be ready to land.